core/io/xml_parser.cpp

```diff
@@ -32,12 +32,18 @@
 
 #include "core/string/print_string.h"
 
+#include <ctype.h>
+
 //#define DEBUG_XML
 
 static inline bool _is_white_space(char c) {
 	return (c == ' ' || c == '\t' || c == '\n' || c == '\r');
 }
 
+static inline bool _is_punctuation(char c) {
+	return ((c == ',') || (c == "’") || (c == '\'') || (c == '"') || (c == '?') || (c == '!') || (c == "-") || (c == "—") || (c == '(') || (c == '[') || (c == '{') || (c == '.') || (c == ':') || (c == ';'));
```

Comment:
`c == "’"` and `c == "—"`: these are Unicode chars outside of ASCII range, since this is UTF-8 stream these won't be a singe char (will be 0xE2 0x80 0x99 and 0xE2 0x80 0x94), and you can't compare it like this.

Reply:
Thank you for your feedback, you are right, in fact I'm now committing an edit for this that should solve the problem